samples/authenticator/src/main/java/org/forgerock/authenticator/sample/view/activity/PushNotificationActivity.java

```diff
@@ -30,18 +52,239 @@ public void onCreate(Bundle savedInstanceState) {
+    }
+
+    private void setupPushBiometricActions(Activity thisActivity, PushNotification notification) {
+        acceptButtons.setVisibility(View.VISIBLE);
+        challengeButtons.setVisibility(View.GONE);
+        rejectChallengeButton.setVisibility(View.GONE);
+
+        final AppCompatActivity activity = this;
+        Button approveButton = findViewById(R.id.pushApprove);
+        approveButton.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.M)
+            @Override
+            public void onClick(View v) {
+                notification.accept(null, null, true, activity, new FRAListener<Void>() {
+                    @Override
+                    public void onSuccess(Void result) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, String.format(getString(R.string.push_notification_approval), "approved"), Toast.LENGTH_LONG).show();
+                                finish();
+                            }
+                        });
+                    }
+
+                    @Override
+                    public void onException(Exception e) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, e.getLocalizedMessage(), Toast.LENGTH_LONG).show();
+                                finish();
+                            }
+                        });
+                    }
+                });
+            }
+        });
+
+        Button denyButton = findViewById(R.id.pushDeny);
+        denyButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                notification.deny(new FRAListener<Void>() {
+                    @Override
+                    public void onSuccess(Void result) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, String.format(getString(R.string.push_notification_approval), "denied"), Toast.LENGTH_LONG).show();
+                                finish();
+                            }
+                        });
+                    }
+
+                    @Override
+                    public void onException(Exception e) {
+                        PushNotificationActivity.this.runOnUiThread(new Runnable() {
+                            @Override
+                            public void run() {
+                                Toast.makeText(thisActivity, R.string.notification_error_network_failure_message, Toast.LENGTH_LONG).show();
```

Comment:
Update the toast message to display the actual error - i.e. e.getLocalizedMessage()